addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -573,7 +579,7 @@ registry.Parallax = Animation.extend({
         // Reset offset if parallax effect will not be performed and leave
         var noParallaxSpeed = (this.speed === 0 || this.speed === 1);
         if (noParallaxSpeed) {
-            this.$bg.css({
+            this._updateBgCss({
```

Comment:
Strange stuff, why do we need that if we restore on destroy? Deserves a comment to say it is by compatibility?

Reply:
This is probably not needed anymore but yeah, I'd keep it to not take any risk, added comment

Reply:
I am not sure to understand. What would be the migration script? To remove the css previously and wrongly saved in the DOM (transform/top/bottom)? If yes, no it's not needed: it doesn't really mess with anything as those are recomputed on page load and then on every page scroll (basically as before this PR). Also, any entering edit mode and save will remove those after this PR.
Doesn't seems worth a script?

I also don't understand the "keep for compatibility", why would we need to reset the css attributes? A short-circuit return should be enough to not go further, no?
I am feeling like I am definitely missing something / the point, please let me know :+1: 

-------------------------------------

About the `TODO`, I actually don't like the opposite (multi line without next lines indented).
I don't see clear guidelines anywhere in any language about that tho, except:
- [Jetbrains](https://stackoverflow.com/questions/22066879/is-it-possible-to-use-multiline-todos-in-intellij-idea): indent one char
```js
// todo: i am a first line
//  i am the second line of the todo
//  i am the third line of the todo
```
- [pycharm](https://stackoverflow.com/questions/48645942/how-do-i-format-a-multi-line-todo-comment-in-pycharm): indent one char too or as I did (after the `TODO:` for all lines)
```py
# todo: i am a first line
#       i am the second line of the todo
#       i am the third line of the todo
```
And in our code, 99% are mono line comments, which is surely the way to go whenever possible. The rest is a mix between no indent and the same indent as I did so no luck there either.

At the end, my rationale is just: how would we detect the end of the TODO comment and the next comment if those are not indented? We could add a blank line but would be weird (would not look like it's linked to the next codeline then)

Eg:
```py
# TODO: It's missing xyz blabla. In master we blabla
#       to be able to know which of the balbla are blabla.
#       But we can't do that in stable.
# This need to be done before xyz so the invoice will be xyz correctly
some_python_line()
```
While with
```py
# TODO: It's missing xyz blabla. In master we blabla
# to be able to know which of the balbla are blabla.
# But we can't do that in stable.
# This need to be done before xyz so the invoice will be xyz correctly
some_python_line()
```
You have no visual clue that the last line is not part of the TODO.

Just sharing my 2cents tho

Reply:
> it's not needed: it doesn't really mess with anything as those are recomputed

> on page load

Well yes, only if we keep that compatibility line, which is the whole point here.

> and then on every page scroll

Not in the case of the `if` we are talking about it, which is also the point: in case wrong CSS was saved, the lines we are commenting here are the only ones which resets the wrong configuration outside of edit mode.

> Doesn't seems worth a script?

That's what I said yes: "Or arguably just remove it in master and not care more about it" :wink: 

> I also don't understand the "keep for compatibility", why would we need to reset the CSS attributes? A short-circuit return should be enough to not go further, no?

I don't know what to tell you ^^ With CSS, the DOM is visually different than without... so if you stop resetting the CSS attributes, this is visually different :shrug: So if you keep the line here, this is only for stable compatibility for the cases of existing parallax that were not properly saved (saved with their inline style CSS).

> how would we detect the end of the TODO comment and the next comment if those are not indented?

In general, just reading the comment which is what they are made for ^^"

> We could add a blank line but would be weird

That's what I do if this is really unclear without it, or in general in case of documentation comment. But anyways, I was not asking you to change that :wink: Was just saying that a mono-line comment in this case would be clearer: this is to remove in master, only kept for compatibility. If you don't agree, that's fine as it is, you'll end up removing those lines in master anyway :laughing:

Reply:
I am still unsure of why are those lines needed, even before my fix, so I'll just add the compatibility comment :innocent: 

Indeed, if `data-scroll-background-ratio` is equal to 0, the options should have removed the parallax bg element (it basically cancel the "top to bottom effect". And it can't reach 1 (grep `Make sure this cannot land on 1`), which are the condition to enter this condition to reset the bg position.

Maybe it was already some compatibility lines, which are useless in Odoo 15 apart from the compatibility.

In either cases, I'll add the comment

Reply:
> And it can't reach 1 (grep Make sure this cannot land on 1), which are the condition to enter this condition to reset the bg position.

It is 1 for all parallax which are in fixed mode, which is the most common effect...
![image](https://github.com/odoo/odoo/assets/10338094/f3ad7e98-51c2-4863-b628-1eac6490b1a0)

> I am still unsure of why are those lines needed, even before my fix

Before your fix, those are probably the lines that resets the background position when you go from one parallax effect to the "fixed" one. That code was wrongly doing the job of an option (at least I imagine it does, not tested). Maybe more?

After your fix, they may not be needed in normal cases... exactly because you reset the background at destroy instead of at parallax effect switching on the option side. Still,I don't care much more than this: we are in stable, we have code that says "On page load, if the parallax effect is 'Fixed', then reset its background position". In stable I would not risk changing that otherwise it means that if an user somehow has a "Fixed" parallax whose bg position was not removed (either by our code or a custo), it will not be removed on page load anymore if you remove those lines.

I actually don't understand why you would want to remove those in 15.0... we keep code that is far less likely to break anything if removed in stable in general :shrug: Especially when it is something that touches on visitor code and/or snippet design. Even if it was `if (speed == 428923)` I would not remove it in stable :grimacing:

Reply:
Just to be clear: I don't want to remove those, it would have 0 value at all in stable, as I mentioned in the first comment:
![image](https://github.com/odoo/odoo/assets/30048408/1f3990d4-044e-4a9e-821c-7398f06396bf)

I just really wanted to get the point of why are those needed :)

> Before your fix, those are probably the lines that resets the background position when you go from one parallax effect to the "fixed" one.

Looks like it indeed

Reply:
My bad then, got confused into thinking you don't see how this could break snippets when removing this (even in master, but there it's ok, we remove the compatibility without caring about migration I'd say for this one).